packages/tools/client-debugger/client-debugger-view/src/components/DataObjectsView.tsx

```diff
@@ -2,39 +2,87 @@
  * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
  * Licensed under the MIT License.
  */
+import {
+	ContainerDataMessage,
+	handleIncomingMessage,
+	HasContainerId,
+	InboundHandlers,
+	ISourcedDebuggerMessage,
+} from "@fluid-tools/client-debugger";
 import React from "react";
+import { useMessageRelay } from "../MessageRelayContext";
 
-import { HasClientDebugger } from "../CommonProps";
 import { SharedObjectRenderOptions } from "../RendererOptions";
 import { DynamicDataView } from "./data-object-views";
 
 /**
  * {@link DataObjectsView} input props.
  */
-export interface DataObjectsViewProps extends HasClientDebugger {
+export interface DataObjectsViewProps extends HasContainerId {
 	/**
 	 * {@inheritDoc RendererOptions}
 	 */
 	renderOptions: SharedObjectRenderOptions;
 }
 
 /**
- * View containing a drop-down style view of {@link DataObjectsViewProps.initialObjects}.
+ * Displays the data inside a container.
  *
  * @remarks
  *
  * Dispatches data object rendering based on those provided view {@link DataObjectsViewProps.renderOptions}.
  */
 export function DataObjectsView(props: DataObjectsViewProps): React.ReactElement {
-	const { clientDebugger, renderOptions } = props;
+	const { containerId, renderOptions } = props;
 
-	const { containerData } = clientDebugger;
+	const messageRelay = useMessageRelay();
+
+	const [containerData, setContainerData] = React.useState<unknown | undefined>();
+
+	React.useEffect(() => {
+		/**
+		 * Handlers for inbound messages from the registry.
+		 */
+		const inboundMessageHandlers: InboundHandlers = {
+			["CONTAINER_DATA"]: (untypedMessage) => {
+				const message = untypedMessage as ContainerDataMessage;
+				if (message.data.containerId === containerId) {
+					setContainerData(message.data.containerData);
+					return true;
+				}
+				return false;
+			},
+		};
+
+		/**
+		 * Event handler for messages coming from the webpage.
+		 */
+		function messageHandler(message: Partial<ISourcedDebuggerMessage>): void {
+			handleIncomingMessage(message, inboundMessageHandlers, {
+				context: "ContainerDataView", // TODO: Fix
+			});
+		}
+
+		messageRelay.on("message", messageHandler);
+
+		// Request state info for the newly specified containerId
```

Comment:
Should we call `setContainerData(undefined) here to prevent display of stale data while we load the appropriate container's data from the debugger?